Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -1133,11 +1141,31 @@ static _WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
 {
     ASSERT(isLoaded());
 
+    BOOL uniqueIdentifierHasChanged = ![m_uniqueIdentifier isEqualToString:lastSeenUniqueIdentifier()];
+    if (uniqueIdentifierHasChanged)
+        saveUniqueIdentifierToStorage();
```

Comment:
The unique identifier in WebKit is not what it was in Safari. If the unique identifier changes in WebKit, all state will be reset, so there is no need to save this here.